bigquery_etl/cli/query.py

```diff
@@ -534,3 +537,213 @@ def initialize(name, sql_dir, project_id, dry_run):
                     default_dataset=f"{project}.{dataset}",
                 )
                 client.query(init_sql, job_config=job_config).result()
+
+
+@query.group(help="Commands for managing query schemas.")
+def schema():
+    """Create the CLI group for the query schema command."""
+    pass
+
+
+@schema.command(
+    help="Update the query schema",
+)
+@click.argument("name")
+@sql_dir_option
+@click.option(
+    "--project-id",
+    "--project_id",
+    help="GCP project ID",
+    default="moz-fx-data-shared-prod",
+    callback=is_valid_project,
+)
+def update(name, sql_dir, project_id):
+    """CLI command for generating the query schema."""
+    if not is_authenticated():
+        click.echo(
+            "Authentication to GCP required. Run `gcloud auth login` "
+            "and check that the project is set correctly."
+        )
+        sys.exit(1)
+    query_files = _queries_matching_name_pattern(name, sql_dir, project_id)
+
+    for query_file in query_files:
+        query_file_path = Path(query_file)
+        query_schema = Schema.from_query_file(query_file_path)
+        existing_schema_path = query_file_path.parent / SCHEMA_FILE
+
+        table_name = query_file_path.parent.name
+        dataset_name = query_file_path.parent.parent.name
+        project_name = query_file_path.parent.parent.parent.name
+        table_schema = Schema.for_table(project_name, dataset_name, table_name)
+
+        if existing_schema_path.is_file():
+            existing_schema = Schema.from_schema_file(existing_schema_path)
+            existing_schema.merge(table_schema)
+            existing_schema.merge(query_schema)
+            existing_schema.to_yaml_file(existing_schema_path)
+        else:
+            query_schema.merge(table_schema)
+            query_schema.to_yaml_file(existing_schema_path)
+
+        click.echo(f"Schema {existing_schema_path} updated.")
+
+        # update bigquery metadata
+        try:
+            client = bigquery.Client()
+            table = client.get_table(f"{project_name}.{dataset_name}.{table_name}")
+            metadata = Metadata.of_query_file(str(query_file_path))
+            metadata_file_path = query_file_path.parent / METADATA_FILE
+
+            if table.time_partitioning and metadata.bigquery.partition_by is None:
+                metadata.set_bigquery_partitioning(
+                    field=table.time_partitioning.field,
+                    partition_type=table.time_partitioning.type_.lower(),
+                    required=table.time_partitioning.require_partition_filter,
+                )
+                click.echo(f"Partitioning metadata added to {metadata_file_path}")
+
+            if table.clustering_fields and metadata.bigquery.cluster_by is None:
+                metadata.set_bigquery_clustering(table.clustering_fields)
+                click.echo(f"Clustering metadata added to {metadata_file_path}")
```

Comment:
I like this approach. We add the partitioning/clustering to metadata if not already set, but otherwise expect the metadata.yaml to be the source of truth for this info.